app/models/patient.rb

```diff
@@ -711,21 +711,59 @@ def latest_transfer
   #  - not in continuous exposure
   #     AND
   #  - on the last day of or past their monitoring period
-  scope :close_eligible, lambda {
-    where(isolation: false)
-      .non_reporting
-      .no_recent_activity
-      .or(
-        exposure_asymptomatic
-        .submitted_assessment_today
-        .end_of_monitoring_period
-      )
+  scope :close_eligible, lambda { |reason = nil|
+    base_scope = exposure_asymptomatic
+                 .submitted_assessment_today
+                 .end_of_monitoring_period
+
+    case reason
+    when nil
+      base_scope.or(no_recent_activity)
+    when :completed_monitoring
+      base_scope
+    when :no_recent_activity
+      no_recent_activity
+    when :enrolled_last_day_monitoring_period
+      base_scope.enrolled_last_day_monitoring_period
+    when :enrolled_past_monitioring_period
+      base_scope.enrolled_past_monitoring_period
+    else
+      throw Exception.new('Invalid reason provided to close_eligible scope!')
+    end
   }
 
   # If a patient record has been inactive for 30 days or more,
   # then it should be automatically closed as part of the close patients job.
   scope :no_recent_activity, lambda {
-    where('updated_at <= ?', 30.days.ago)
+    where(isolation: false)
+      .non_reporting
```

Comment:
Why the addition on `non_reporting`?

Reply:
I was just moving logic around that was already there. I actually remove this helper scope completely in my next commit